Tests/MongoSwiftTests/SpecTestRunner/Match.swift

```diff
@@ -47,38 +71,38 @@ extension Array: Matchable {
 
 /// Extension that adds MATCHES functionality to `Document`.
 extension Document: Matchable {
-    internal func contentMatches(expected: Any) -> Bool {
-        guard !isPlaceholder(expected) else {
-            return true
-        }
-
-        guard let expected = expected as? Document else {
-            return false
-        }
-
+    internal func contentMatches(expected: Document) -> Bool {
         for (eK, eV) in expected {
-            guard let aV = self[eK], aV.bsonMatches(expected: eV) else {
+            guard let aV = self[eK], aV.matches(expected: eV) else {
                 return false
             }
         }
         return true
     }
 }
 
-/// Extension that adds MATCHES functionality to `BSONValue`.
-extension BSONValue {
-    internal func bsonMatches(expected: BSONValue) -> Bool {
-        if let matchable = self as? Matchable {
-            return matchable.matches(expected: expected)
+/// Extension that adds MATCHES functionality to `BSON`.
+extension BSON: Matchable {
+    internal func isPlaceholder() -> Bool {
+        return self.asInt() == 42 || self.stringValue == "42"
```

Comment:
rather than hard-coding `42` and `"42"` in twice, maybe it would be better to just check whether `self.asInt().isPlaceholder() || self.stringValue?.isPlaceholder()`?

Reply:
good idea, updated.